pkg/scheduler/framework/interface.go

```diff
@@ -160,6 +162,33 @@ func (s *Status) AsError() error {
 	return errors.New(s.Message())
 }
 
+// IsStatusEqual returns whether two Statuses are equal to each other.
+func IsStatusEqual(statusX, statusY *Status) bool {
+	if statusX == statusY {
+		return true
+	}
+	if statusX == nil {
+		return statusY == nil || statusY.code == Success
+	}
+	if statusY == nil {
+		return statusX.code == Success
+	}
+	if statusX.code != statusY.code {
+		return false
+	}
+	if statusX.err == nil && statusY.err == nil {
+		if !cmp.Equal(statusX.reasons, statusY.reasons) {
+			return false
+		}
+	}
+	if statusX.err != nil && statusY.err != nil {
+		if cmp.Equal(statusX.err.Error(), statusY.err.Error()) {
+			return true
+		}
+	}
```

Comment:
I suppose this is included in the line below? (cmpopts.EquateErrors())

Reply:
I think below case should be true while cmpopts.EquateErrors() will take it as false.

```
			x:    NewStatus(Error, "error"),
			y:    NewStatus(Error, "error"),
```

Reply:
Further more, what about 

```
			x:    NewStatus(Error, "error"),
			y:    AsStatus(fmt.Errorf("error: %w, NewStatus(Error, "error").AsError())),
```

Reply:
Rethink about this, just follow the behavior of `cmpopts.EquateErrors()` SGTM. We should take error status with same error string as not equal. If so, we maybe better to assume other status only equal when they are the same ones.

```
	x := fmt.Errorf("test")
	y := fmt.Errorf("test")
	d := cmp.Diff(x, y, cmpopts.EquateErrors())  ## d != ""
```

Reply:
I agree. We should be exporting the root errors when necessary to do precise checks. Or have the errors implement Is